zcash_primitives/src/transaction/components/orchard.rs

```diff
@@ -10,10 +12,50 @@ use orchard::{
     Anchor,
 };
 
+use super::Amount;
+use crate::serialize::{Array, CompactSize};
+use crate::{serialize::Vector, transaction::Transaction};
+
 pub const FLAG_SPENDS_ENABLED: u8 = 0b0000_0001;
 pub const FLAG_OUTPUTS_ENABLED: u8 = 0b0000_0010;
 pub const FLAGS_EXPECTED_UNSET: u8 = !(FLAG_SPENDS_ENABLED | FLAG_OUTPUTS_ENABLED);
 
+/// Reads an [`orchard::Bundle`] from a v5 transaction format.
+pub fn read_v5_bundle<R: Read>(
+    mut reader: R,
+) -> io::Result<Option<orchard::Bundle<Authorized, Amount>>> {
+    let actions_without_auth = Vector::read(&mut reader, |r| read_action_without_auth(r))?;
```

Comment:
You have to disable the clippy lint for redundant closures here; it fails typechecking on a lifetime issue if you try to remove the explicit `r`.